integration-tests/smoke/ccip_test.go

```diff
@@ -168,105 +151,233 @@ func TestTokenTransfer(t *testing.T) {
+func getTokenByAddress(dstToken1, dstToken2 *burn_mint_erc677.BurnMintERC677, tokenAddress common.Address) *burn_mint_erc677.BurnMintERC677 {
+	if dstToken1.Address() == tokenAddress {
+		return dstToken1
+	} else if dstToken2.Address() == tokenAddress {
+		return dstToken2
+	}
+	return nil
+}
 
-	// After commit is reported on all chains, token prices should be updated in FeeQuoter.
-	for dest := range e.Chains {
-		linkAddress := state.Chains[dest].LinkToken.Address()
-		feeQuoter := state.Chains[dest].FeeQuoter
-		timestampedPrice, err := feeQuoter.GetTokenPrice(nil, linkAddress)
-		require.NoError(t, err)
-		require.Equal(t, ccdeploy.MockLinkPrice, timestampedPrice.Value)
+// Helper function to determine the destination token
+func getDestinationToken(scenario struct {
+	name         string
+	srcChain     uint64
+	dstChain     uint64
+	tokenAmounts []router.ClientEVMTokenAmount
+	receiver     common.Address
+	data         []byte
+}, tenv ccdeploy.DeployedEnv, srcToken1, dstToken1, srcToken2, dstToken2 *burn_mint_erc677.BurnMintERC677, tokenAddress common.Address) *burn_mint_erc677.BurnMintERC677 {
+	if scenario.srcChain == tenv.HomeChainSel && scenario.dstChain == tenv.FeedChainSel {
+		if tokenAddress == srcToken1.Address() {
+			return dstToken1
+		} else if tokenAddress == srcToken2.Address() {
+			return dstToken2
+		}
+	} else if scenario.srcChain == tenv.FeedChainSel && scenario.dstChain == tenv.HomeChainSel {
+		if tokenAddress == dstToken1.Address() {
+			return srcToken1
+		} else if tokenAddress == dstToken2.Address() {
+			return srcToken2
+		}
 	}
+	return nil
+}
 
-	// Wait for all exec reports to land
-	ccdeploy.ConfirmExecWithSeqNrForAll(t, e, state, expectedSeqNum, startBlocks)
+func setupEnvironment(t *testing.T) (ccdeploy.DeployedEnv, ccdeploy.CCIPOnChainState) {
+	lggr := logger.TestLogger(t)
+	tenv, _, _ := testsetups.NewLocalDevEnvironmentWithDefaultPrice(t, lggr)
+	state, err := ccdeploy.LoadOnchainState(tenv.Env)
+	require.NoError(t, err)
+	return tenv, state
+}
+
+func deployAndApproveTokens(t *testing.T, e ccdeploy.DeployedEnv, state ccdeploy.CCIPOnChainState, tokenName string) (*burn_mint_erc677.BurnMintERC677, *burn_mint_erc677.BurnMintERC677) {
+	srcToken, _, dstToken, _, err := ccdeploy.DeployTransferableToken(
+		logger.TestLogger(t),
+		e.Env.Chains,
+		e.HomeChainSel,
+		e.FeedChainSel,
+		state,
+		e.Env.ExistingAddresses,
+		tokenName,
+	)
+	require.NoError(t, err)
+
+	tenTokens := new(big.Int).Mul(big.NewInt(1e18), big.NewInt(10))
+	mintAndApprove(t, e, state, srcToken, e.HomeChainSel, tenTokens)
+	mintAndApprove(t, e, state, dstToken, e.FeedChainSel, tenTokens)
 
-	balance, err := dstToken.BalanceOf(nil, state.Chains[tenv.FeedChainSel].Receiver.Address())
+	return srcToken, dstToken
+}
+
+func mintAndApprove(t *testing.T, e ccdeploy.DeployedEnv, state ccdeploy.CCIPOnChainState, token *burn_mint_erc677.BurnMintERC677, chainSel uint64, amount *big.Int) {
```

Comment:
I think we have a very similar helper for the USDC, it would be great to dedup these 
https://github.com/smartcontractkit/chainlink/blob/5a686748165e25c39f6e97d5c2141777440d508d/integration-tests/smoke/ccip_usdc_test.go#L254